File: 0290_Word Pattern/WordPattern.java
class Solution {
    public boolean wordPattern(String pattern, String s) {
        HashMap mapToIndex = new HashMap();
        String[] words = s.split(" ");

        if (words.length != pattern.length())
            return false;

        for (Integer i = 0; i < words.length; i++) {
            char c = pattern.charAt(i);
            String word = words[i];

            if (!mapToIndex.containsKey(c))
                mapToIndex.put(c, i);

            if (!mapToIndex.containsKey(word))
                mapToIndex.put(word, i);

            if (mapToIndex.get(c) != mapToIndex.get(word))
                return false;
        }

        return true;
    }
}

File: 0290_Word Pattern/WordPattern.cs
public class Solution {
    public bool WordPattern(string pattern, string s) {
        var words = s.Split(' ');
        if(pattern.Length != words.Count()) return false;
        
        var dictChar = new Dictionary<char, string>();
        var dictWord = new Dictionary<string, char>();
        
        for(int i=0;i<pattern.Length;i++)
        {
            var c = pattern[i];
            var word = words[i];
            
            if(!dictChar.ContainsKey(c))
            {
                if(dictWord.ContainsKey(word)) return false;
                
                dictChar.Add(c, word);
                dictWord.Add(word, c);

            }else{
                var mappedWord = dictChar[c];
                if(!mappedWord.Equals(word)) return false;
            }
        }
        
        return true;
    }
}